util/conditions/experimental/merge_strategies.go

```diff
@@ -0,0 +1,408 @@
+}
+
+// getPriority returns the merge priority for each condition.
+// The default implementation identifies as:
+// - issues: conditions with positive polarity (normal True) and status False or conditions with negative polarity (normal False) and status True.
+// - unknown: conditions with status unknown.
+// - info: conditions with positive polarity (normal True) and status True or conditions with negative polarity (normal False) and status False.
+func getPriority(condition metav1.Condition, negativePolarityConditionTypes sets.Set[string]) mergePriority {
+	switch condition.Status {
+	case metav1.ConditionTrue:
+		if negativePolarityConditionTypes.Has(condition.Type) {
+			return issueMergePriority
+		}
+		return infoMergePriority
+	case metav1.ConditionFalse:
+		if negativePolarityConditionTypes.Has(condition.Type) {
+			return infoMergePriority
+		}
+		return issueMergePriority
+	case metav1.ConditionUnknown:
+		return unknownMergePriority
+	}
+
+	// Note: this should never happen. In case, those conditions are considered like condition with unknown status.
+	return unknownMergePriority
+}
+
+// aggregateMessages return messages for the aggregate operation.
+func aggregateMessages(conditions []ConditionWithOwnerInfo, n *int, kind string, dropEmpty bool) (messages []string, other int) {
+	// create a map with all the messages and the list of objects reporting the same message.
+	messageObjMap := map[string][]string{}
+	for _, condition := range conditions {
+		if dropEmpty && condition.Message == "" {
+			continue
+		}
+
+		m := fmt.Sprintf("(%s)", condition.Status)
+		if condition.Message != "" {
+			m += fmt.Sprintf(": %s", condition.Message)
+		}
+		if _, ok := messageObjMap[m]; !ok {
+			messageObjMap[m] = []string{}
+		}
+		messageObjMap[m] = append(messageObjMap[m], klog.KRef(condition.OwnerResource.Namespace, condition.OwnerResource.Name).String())
+	}
+
+	// compute the order of messages according to the number of objects reporting the same message.
+	// Note: The message text is used as a secondary criteria to sort messages with the same number of objects.
+	messageIndex := make([]string, 0, len(messageObjMap))
+	for m := range messageObjMap {
+		messageIndex = append(messageIndex, m)
+	}
+
+	sort.SliceStable(messageIndex, func(i, j int) bool {
+		return len(messageObjMap[messageIndex[i]]) > len(messageObjMap[messageIndex[j]]) ||
+			(len(messageObjMap[messageIndex[i]]) == len(messageObjMap[messageIndex[j]]) && messageIndex[i] < messageIndex[j])
+	})
+
+	// Pick the first n messages, decrement n.
+	// For each message, add up to three objects; if more add the number of the remaining objects with the same message.
+	// Count the number of objects reporting messages not included in the above.
+	// Note: we are showing up to three objects because usually control plane has 3 machines, and we want to show all issues
+	// to control plane machines if any,
+	for i := range len(messageIndex) {
```

Comment:
```suggestion
	for _, m := range messageIndex {
```

\+ use m instead of messageIndex[i] everywhere in this func

Would this also work? seems simpler

Reply:
fixed, it is slightly better